Private networks in gateway resolver (#25178)
cloudflare/cloudflare-docs: src/content/changelog/gateway/2025-09-11-dns-filtering-for-private-network-onramps.mdx

---

> title: DNS filtering for private network onramps.

Reviewer: ```suggestion
title: DNS filtering for private network onramps
```

---
title: DNS filtering for private network onramps.

> description: Magic WAN and WARP Connector traffic can now route DNS queries privately to Gateway Resolver without public internet exposure.

Reviewer: ```suggestion
description: Magic WAN and WARP Connector traffic can now privately route DNS queries to the Gateway resolver without public Internet exposure.
```

products:
  - gateway
  - magic-wan
  - cloudflare-tunnel
date: "2025-09-11"
---

[Magic WAN](/magic-wan/zero-trust/cloudflare-gateway/#dns-filtering) and [WARP Connector](/cloudflare-one/connections/connect-networks/private-net/warp-connector/site-to-internet/#configure-dns-resolver-on-devices) customers can now securely route their DNS traffic to Gateway Resolver without exposing traffic to the public internet.

> This feature enables DNS resolution and filtering for traffic coming from private networks while preserving source Internal IP visibility.
> This ensures Magic WAN customers have full integration with our Cloudflare One features including Internal DNS and hostname-based policies.

Reviewer: ```suggestion
Routing DNS traffic to the Gateway resolver allows DNS resolution and filtering for traffic coming from private networks while preserving source internal IP visibility. This ensures Magic WAN users have full integration with our Cloudflare One features, including [Internal DNS](/cloudflare-one/policies/gateway/resolver-policies/#internal-dns) and [hostname-based policies](/cloudflare-one/policies/gateway/egress-policies/#selector-prerequisites).
```
